homeassistant/components/netatmo/climate.py

```diff
@@ -3,71 +3,132 @@
+    STATE_HEAT, STATE_IDLE, SUPPORT_ON_OFF, SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_OPERATION_MODE, SUPPORT_AWAY_MODE, STATE_MANUAL, STATE_AUTO,
+    STATE_ECO, STATE_COOL)
 from homeassistant.util import Throttle
 import homeassistant.helpers.config_validation as cv
 
+
 DEPENDENCIES = ['netatmo']
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_RELAY = 'relay'
-CONF_THERMOSTAT = 'thermostat'
+CONF_HOMES = 'homes'
+CONF_ROOMS = 'rooms'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
 
-DEFAULT_AWAY_TEMPERATURE = 14
-# # The default offset is 2 hours (when you use the thermostat itself)
-DEFAULT_TIME_OFFSET = 7200
-# # Return cached results if last scan was less then this time ago
-# # NetAtmo Data is uploaded to server every hour
-MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=300)
+HOME_CONFIG_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Optional(CONF_ROOMS, default=[]): vol.All(cv.ensure_list, [cv.string])
+})
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_RELAY): cv.string,
-    vol.Optional(CONF_THERMOSTAT, default=[]):
-        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_HOMES): vol.All(cv.ensure_list, [HOME_CONFIG_SCHEMA])
 })
 
+STATE_NETATMO_SCHEDULE = 'schedule'
+STATE_NETATMO_HG = 'hg'
+STATE_NETATMO_MAX = 'max'
+STATE_NETATMO_AWAY = 'away'
+STATE_NETATMO_OFF = STATE_OFF
+STATE_NETATMO_MANUAL = STATE_MANUAL
+
+DICT_NETATMO_TO_HA = {
+    STATE_NETATMO_SCHEDULE: STATE_AUTO,
+    STATE_NETATMO_HG: STATE_COOL,
+    STATE_NETATMO_MAX: STATE_HEAT,
+    STATE_NETATMO_AWAY: STATE_ECO,
+    STATE_NETATMO_OFF: STATE_OFF,
+    STATE_NETATMO_MANUAL: STATE_MANUAL
+}
+
+DICT_HA_TO_NETATMO = {
+    STATE_AUTO: STATE_NETATMO_SCHEDULE,
+    STATE_COOL: STATE_NETATMO_HG,
+    STATE_HEAT: STATE_NETATMO_MAX,
+    STATE_ECO: STATE_NETATMO_AWAY,
+    STATE_OFF: STATE_NETATMO_OFF,
+    STATE_MANUAL: STATE_NETATMO_MANUAL
+}
+
 SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE |
                  SUPPORT_AWAY_MODE)
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the NetAtmo Thermostat."""
     netatmo = hass.components.netatmo
-    device = config.get(CONF_RELAY)
 
     import pyatmo
+    homes_conf = config.get(CONF_HOMES)
     try:
-        data = ThermostatData(netatmo.NETATMO_AUTH, device)
-        for module_name in data.get_module_names():
-            if CONF_THERMOSTAT in config:
-                if config[CONF_THERMOSTAT] != [] and \
-                   module_name not in config[CONF_THERMOSTAT]:
-                    continue
-            add_entities([NetatmoThermostat(data, module_name)], True)
+        home_data = HomeData(netatmo.NETATMO_AUTH)
     except pyatmo.NoDevice:
-        return None
+        return
+
+    homes = []
+    rooms = {}
+    if homes_conf is not None:
+        for home_conf in homes_conf:
+            home = home_conf.get(CONF_NAME)
```

Comment:
This is required so you should use `home_conf[CONF_NAME]` (brace syntax).

Reply:
Resolved in 4515f57.